lib/jsdom/living/helpers/geometry/matrixDomStringParser.js

```diff
@@ -0,0 +1,397 @@
+          this.advance();
+          this.advance();
+          break;
+        }
+      }
+    }
+  }
+
+  getFunction() {
+    let buffer = "";
+
+    while (!this.is(SYMBOL.lparen)) {
+      buffer += this.currentChar;
+      this.advance();
+    }
+
+    // discard lparen
+    this.advance();
+
+    return buffer;
+  }
+
+  getParams() {
+    let buffer = "";
+    const params = [];
+
+    while (!this.is(SYMBOL.rparen)) {
+      if (
+        !this.is(SYMBOL.whitespace) &&
+        !this.is(SYMBOL.comma)
+      ) {
+        buffer += this.currentChar;
+      } else if (buffer.length > 0) {
+        params.push(buffer);
+        buffer = "";
+      }
+
+      this.advance();
+    }
+
+    // clear remaining buffer
+    if (buffer.length > 0) {
+      params.push(buffer);
+    }
+
+    // discard rparen
+    this.advance();
+
+    return params;
+  }
+
+  parse() {
+    const functions = [];
+
+    while (this.index < this.buffer.length - 1) {
+      this.skipWhiteSpace();
+      this.skipComment();
+
+      const transformFunction = this.getFunction();
+      const transformParams = this.getParams();
+
+      functions.push([transformFunction, transformParams]);
+    }
+
+    return functions;
+  }
+}
+
+function validateParameterLength(expectedLength, params, fnName = "") {
+  const isValidLength = Array.isArray(expectedLength) ?
+    params.length >= expectedLength[0] && params.length <= expectedLength[1] :
+    params.length === expectedLength;
+
+  if (!isValidLength) {
+    throw new TypeError(`Received invalid number of parameters for function ${fnName}`);
+  }
+}
+
+// https://drafts.csswg.org/css-transforms-1/#typedef-transform-function
+// https://drafts.csswg.org/css-transforms-2/#three-d-transform-functions
+const transformFunctionNames = [
+  "matrix3d",
+  "matrix",
+  "translate3d",
+  "translate",
+  "translateX",
+  "translateY",
+  "translateZ",
+  "scale3d",
+  "scale",
+  "scaleX",
+  "scaleY",
+  "scaleZ",
+  "rotate3d",
+  "rotate",
+  "rotateX",
+  "rotateY",
+  "rotateZ",
+  "perspective",
+  "skew",
+  "skewX",
+  "skewY"
+];
+// https://www.w3.org/TR/css-values-4/#absolute-length
+const absoluteLengths = ["cm", "mm", "Q", "in", "pc", "pt", "px"];
+const LENGTH_CONVERTERS = {
+  cm(x) {
+    return this.in(x) / 2.54;
+  },
+  mm(x) {
+    return this.cm(x) / 10;
+  },
+  Q(x) {
+    return this.cm(x) / 40;
+  },
+  in(x) {
+    return x * 96;
+  },
+  pc(x) {
+    return this.in(x) / 6;
+  },
+  pt(x) {
+    return this.in(x) / 72;
+  },
+  px(x) {
+    return x;
+  }
+};
+
+// https://drafts.csswg.org/css-values-3/#angles
+const angleUnits = ["deg", "grad", "rad", "turn"];
+const ANGLE_CONVERTERS = {
+  deg(x) {
+    return x;
+  },
+  grad(x) {
+    return this.turn(x) / 400;
+  },
+  rad(x) {
+    return x * (180 / Math.PI);
+  },
+  turn(x) {
+    return x * 360;
+  }
+};
+
+const paramConverter = {
+  _rules: {
+    number(x) {
+      return (SYMBOL.radixPoint.test(x) || !isNaN(x)) && x !== null;
```

Comment:
For one, this doesn't take into account `e`, like `1e3`. I recommend following https://drafts.csswg.org/css-syntax-3/#consume-numeric-token closely.

Reply:
The parser will be closer to spec with the next commit, I just wanted something to get the ball rolling here. This is very much quick and dirty in it's current state